intelhex/test.py

```diff
@@ -1169,6 +1169,17 @@ def test_merge_overlap(self):
         ih2 = IntelHex({0:2})
         ih1.merge(ih2, overlap='replace')
         self.assertEqual({0:2}, ih1.todict())
+        # check: same value: ok
+        ih1 = IntelHex({0:1})
+        ih2 = IntelHex({0:1})
+        ih1.merge(ih2, overlap='check')
+        self.assertEqual({0:1}, ih1.todict())
+        # check: different value: raise error
+        ih1 = IntelHex({0:1})
+        ih2 = IntelHex({0:2})
+        self.assertRaisesMsg(intelhex.AddressOverlapError,
+            'Data at address 0x0 is different: 0x1 vs. 0x2',
+            ih1.merge, ih2, overlap='check')
 
```

Comment:
The test in itself looks legit, but the `test_merge_wrong_args` right above this one now yields an incomplete message if triggered. Please add the new argument into that message and make it work again.

Reply:
Hum, seems that I didn't play the test... Any it's fixed.

I also updated the `docs/`